twoaxistracking/layout.py

```diff
@@ -121,6 +121,58 @@ def generate_field_layout(gcr, total_collector_area, min_tracker_spacing,
     relative_azimuth = np.mod(450-np.rad2deg(np.arctan2(Y, X)), 360)
     # Relative slope of collectors
     # positive means collector is higher than reference collector
-    relative_slope = -np.cos(np.deg2rad(slope_azimuth - relative_azimuth)) * slope_tilt  # noqa: E501
+    relative_slope = np.rad2deg(np.arctan(-np.cos(np.deg2rad(slope_azimuth - relative_azimuth))
+                                          * np.tan(np.deg2rad(slope_tilt))))
 
     return X, Y, Z, tracker_distance, relative_azimuth, relative_slope
+
+
+def get_max_shading_elevation(total_collector_geometry, tracker_distance,
+                              relative_azimuth, relative_slope):
+    """Calculate the maximum elevation angle for which shading can occur.
+
+    Parameters
+    ----------
+    total_collector_geometry: Shapely Polygon
+        Polygon corresponding to the total collector area.
+    tracker_distance: array of floats
+        Distances between neighboring trackers and the reference tracker.
+    relative_azimuth: array of floats
+        Relative azimuth of neighboring trackers - measured clockwise from
+        north [degrees].
+    relative_slope: array of floats
+        Slope between neighboring trackers and reference tracker. A positive
+        slope means neighboring collector is higher than reference collector.
+
+    Returns
+    -------
+    max_shading_elevation: float
+        The highest solar elevation angle for which shading can occur for a
+        given field layout and collector geometry [degrees]
+
+    Note
+    ----
+    The maximum shading elevation angle is calculated for all neighboring
+    trackers and overall maximum is returned. The maximum shading elevation
+    occurs when one of the upper corners of the projeted shading geometry and
+    the lower corner of the reference collector intersects.
+
+    The calculations are based on the bounding box of the collector geometry,
+    thus for non-rectangular geometries the maximum shading elevation is a
+    conservative estimate.
+    """
+    # Calculate extent of box bounding the total collector geometry
+    # Should it be the active_colletor_geometry???
+    x_min, y_min, x_max, y_max = total_collector_geometry.bounds
+    # Collector dimensions
+    x_dim = x_max - x_min
+    y_dim = y_max - y_min
+    delta_gamma_rad = np.arcsin(x_dim / tracker_distance)
+    max_elevations = np.rad2deg(np.arcsin(
+        y_dim * np.cos(np.deg2rad(relative_slope)) /
+        (tracker_distance * np.cos(delta_gamma_rad)))) + relative_slope
+    # Set nan values to 90 degrees (no maximum shading elevation)
+    max_elevations = np.nan_to_num(max_elevations, nan=90)
```

Comment:
I'm having trouble coming up with a valid scenario where there are nans here.  What is this guarding against?

Reply:
I'm still not seeing it.  Maybe add a test that would return NaN without this line?

Reply:
@kanderso-nrel I'm not understanding what you mean by adding a test without the line... the line is necessary for the code to work properly for all geometries.

Reply:
It took me a while, but I get it now. Good idea.